packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -660,56 +687,56 @@ contract BridgeRouter is
         address originator = readRequestToOriginator[operationId];
         if (originator == address(0)) revert InvalidParams();
 
-        // Try to deliver the response
-        bool delivered = false;
+        ICrossChainStateReadReceiver(originator).receiveStateRead(
+            resultData,
+            originator,
+            operationId,
+            sourceChainId
+        );
 
-        // Check if the originator implements the ICrossChainStateReadReceiver interface
-        bytes4 interfaceId = type(ICrossChainStateReadReceiver).interfaceId;
-        try
-            ICrossChainStateReadReceiver(originator).supportsInterface(
-                interfaceId
-            )
-        returns (bool supported) {
-            if (supported) {
-                try
-                    ICrossChainStateReadReceiver(originator).receiveStateRead(
-                        resultData,
-                        originator,
-                        operationId,
-                        sourceChainId
-                    )
-                {
-                    delivered = true;
-                } catch {
-                    delivered = false;
-                }
-            } else {
-                (bool success, ) = originator.call(
-                    abi.encodeWithSelector(
-                        ICrossChainStateReadReceiver.receiveStateRead.selector,
-                        resultData,
-                        originator,
-                        operationId,
-                        sourceChainId
-                    )
-                );
-                delivered = success;
-            }
-        } catch {
-            delivered = false;
-        }
+        emit ReadResponseDelivered(operationId, originator, true);
+    }
+
+    /// @inheritdoc IBridgeRouter
+    function deliver(
```

Comment:
this is not used in any adapter ? We'll do it in separate PR ?

Reply:
Yea, wanted to stay small